### DISCUSSION TITLE
Replace fineuploader with Uppy

### DISCUSSION THREAD

**Discussion by nmlsdev** [mediacms-io/mediacms#787] [Feature Requests]
**Describe the feature you'd like**
As a MediaCMS System Administrator, I'd like to be able to recursively import media files from an existing directory on the OS.

**Describe alternatives you've considered**
Though it could be possible to script an API call looping over each file in an existing directory, it doesn't make sense if the media is already stored on the server itself.

**Additional context**
Information that could potentially be required for media to exist on the server could be asked for on the page before submitting the request for a directory scan, such as "user", "category", etc...

**Comment by srvprivate** [comment]
This NEEDs to be a HIGH priority. With the given examples of companies who use mediaCMS, I'm a video producer who would use this full scale. Having to import all of my media which is itself stored ON the server seems redundant, and I refuse to do that. Please look into scanning existing media databases. I KNOW its possible! API call?

**Reply by mgogoulos** [reply]
I agree with you and would like to find time or resources to work on it. The upload form needs be refactored, because the existing one is based on fine uploader js that  does not support the ability to pass the path on all browsers and is deprecated anyways...An alternative form need be created

**Reply by srvprivate** [reply]
Wonderful! Happy to stay in the loop. I get time and resources….You’ve done brilliant work so far, this is the best I’ve seen. Just wait till if and when you make this federated or something ha! Could out do peer tube.

But yes, this seems like a solid issue to work at! Thanks for the reply!

**Reply by nmlsdev** [reply]
> I agree with you and would like to find time or resources to work on it. The upload form needs be refactored, because the existing one is based on fine uploader js that does not support the ability to pass the path on all browsers and is deprecated anyways...An alternative form need be created

I'd love to help contribute to the development effort of this issue, but I'm a bit new to the python world. What IDE or other development tools would make my life easier in trying to understand and work with this code-base?

**Reply by Dabitch** [reply]
I might be on the wrong track here, but would something like uppy be useful to use instead? Uppy fetches files locally and can also fetch from remote places like Dropbox.

https://github.com/transloadit/uppy

**Reply by mgogoulos** [reply]
Absolutely! Uppy looks to be what we need here.

**Comment by skydudie** [comment]
If you are okay with using Django shell, you can manually trigger an import that will import an entire directory worth of media.

Check out the script here: https://github.com/mediacms-io/mediacms/discussions/960#discussioncomment-9974512

**Reply by Dabitch** [reply]
Very useful, thank you.
